metals/src/main/scala/scala/meta/internal/metals/formatting/IndentOnPaste.scala

```diff
@@ -8,8 +8,13 @@ import org.eclipse.lsp4j.TextEdit
 
 object IndentOnPaste extends RangeFormatter {
 
+  private val noEndClause = raw"((<!\bend\b\s*?)\b(if|while|for|match|try))"
+  private val ifThenClause = raw"(\bif\s+(?!.*?\bthen\b.*?$$)[^\s]*?)"
+  private val keywordClause = raw"\b(then|else|do|catch|finally|yield|case)"
+  private val openBraceClause = raw"""(^.*\{[^}"']*$$"""
```

Comment:
Should it start from `(` or should it have `)` at the end?
Originally it was: `(^.*\{[^}"']*$$)`

Reply:
It should have the complete grouping, so 
```scala
private val openBraceClause = raw"""(^.*\{[^}"']*$$)"""
```

And then the full `increaseIndentPatternRegex` becomes 

```scala
raw"""($noEndClause|$ifThenClause|$keywordClause|$extensionClause|=|=>|<-|=>>|:)\s*?$$|$openBraceClause""".r
```

I believe that the confusion arose from the initial opening bracket that is closed after the arrows matching, with a `:)`, not at the pure end of the regex.

Reply:
Fixed! The only problem with the extension regex is that it will not work with mutiline extension definition, but I think it's fine for now.